semantic_release/vcs_helpers.py

```diff
@@ -1,21 +1,27 @@
 import re
 
-from git import GitCommandError, NoSuchPathError, Repo, TagObject
+from git import GitCommandError, InvalidGitRepositoryError, Repo, TagObject
 
 from .errors import GitError
 from .settings import config
 
 try:
-    repo = Repo('.git', search_parent_directories=True)
-except NoSuchPathError:
+    repo = Repo('.', search_parent_directories=True)
+except InvalidGitRepositoryError:
```

Comment:
Shouldn't `NoSuchPathError`still be here?

Reply:
No, it shouldn't!
I had to change it in order to catch the exception raised by `Repo` in case provided path doesn't lead to a git reporitory: apparently the exception raised is `InvalidGitRepositoryError`.
`NoSuchPathError` is raised only in case you provide an invalid path, which is unlikely for the `'.'` representing the current folder: [see docs](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=Repo#git.exc.NoSuchPathError).
With your version of the code is very likely you get `NoSuchPathError` when there's not `.git` folder from where you ran your command...